src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,230 @@
 package core.basesyntax;
 
+import java.util.Collection;
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> firstNode;
+    private Node<T> lastNode;
+    private int size;
+
     @Override
     public void add(T value) {
+        linkedLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkPositionIndex(index);
+        if (index == size) {
+            linkedLast(value);
+        } else {
+            linkBefore(value, node(index));
+        }
+    }
+
+    void linkBefore(T e, Node<T> succ) {
+        final Node<T> pred = succ.previous;
+        final Node<T> newNode = new Node<>(pred, e, succ);
+        succ.previous = newNode;
+
+        if (pred == null) {
+            firstNode = newNode;
+        } else {
+            pred.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        addAll(size, list);
+    }
+
+    public void addAll(int index, Collection<? extends T> c) {
```

Comment:
Create informative variable and method names. The method name `addAll` does not clearly convey that it adds all elements of a collection starting at a specified index. Consider renaming it to better reflect its functionality.